src/github_query_project_state.py

```diff
@@ -6,73 +6,70 @@
 
-    # TODO: This part of code almost identical as in query_issues.py script
-    # TODO: Make an util method for parsing repositories
-    # Get environment variables from the controller script
-    github_token = os.getenv('GITHUB_TOKEN')
-    repositories_env = os.getenv('REPOSITORIES')
-    projects_title_filter = os.getenv('PROJECTS_TITLE_FILTER')
+    logging.info("Script for downloading project data from GitHub GraphQL started.")
 
-    project_state_mining = os.getenv('PROJECT_STATE_MINING')
-    project_state_mining = project_state_mining.lower() == 'true'
-
-    # Parse repositories JSON string
-    try:
-        repositories_json = json.loads(repositories_env)
-    except json.JSONDecodeError as e:
-        print(f"Error parsing REPOSITORIES: {e}")
-        exit(1)
+    # Load action inputs from the environment
+    action_inputs = ActionInputs().load_from_environment()
 
     # Check if the output directory exists and create it if not
     current_dir = os.path.dirname(os.path.abspath(__file__))
     ensure_folder_exists(OUTPUT_DIRECTORY, current_dir)
 
-    # Initialize the request session
-    session = initialize_request_session(github_token)
-
     # Check if project mining is allowed and exit the script if necessary
-    if not project_state_mining:
-        print("Project data mining is not allowed. The process will not start.")
+    if not action_inputs.is_project_state_mining_enabled:
+        logging.info("Project data mining is not allowed. The process will not start.")
         exit()
 
-    print("Project data mining allowed, starting the process.")
+    logging.info("Project data mining allowed, starting the process.")
+
+    # Initialize GitHub instance
+    GithubManager().initialize_github_instance(action_inputs.github_token, ISSUES_PER_PAGE_LIMIT)
+
+    # Initialize the request session
+    GithubManager().initialize_request_session(action_inputs.github_token)
 
-    projects = {}
+    # Mine project issues for every repository
+    projects = []
+    project_issues = []
+    config_repositories = action_inputs.repositories
+    for config_repository in config_repositories:
+        repository_id = f"{config_repository.owner}/{config_repository.name}"
 
-    # Generate main structure for every unique project, that is attached to any input repository
-    for repository_json in repositories_json:
-        repository = Repository()
-        repository.load_from_json(repository_json)
+        if GithubManager().store_repository(repository_id) is None:
+            return None
 
-        # Update `projects` dictionary with key / value of project_id / project object
-        repository.get_unique_projects(session, projects_title_filter, projects)
+        # Fetch all projects attached to the repository
+        projects_title_filter = action_inputs.projects_title_filter
+        projects.extend(prjts := GithubManager().fetch_repository_projects(projects_title_filter))
```

Comment:
Rename `prjts` to `prjs`, please.

Reply:
Walrus operator is renamed in commit: [fb78190](https://github.com/AbsaOSS/living-doc-generator/pull/16/commits/fb78190e5adedc65ded39abff55adc6f51da5407).